<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>598</width>
    <height>538</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0" colspan="4">
     <widget class="QGraphicsView" name="graphicsView"/>
    </item>
    <item row="1" column="2" colspan="2">
     <widget class="QLineEdit" name="bw_t"/>
    </item>
    <item row="1" column="0">
     <widget class="QPushButton" name="btn_bw">
      <property name="text">
       <string>Black &amp; White</string>
      </property>
     </widget>
    </item>
    <item row="2" column="0">
     <widget class="QPushButton" name="btn_logtrans">
      <property name="text">
       <string>Log_trans</string>
      </property>
     </widget>
    </item>
    <item row="2" column="1">
     <widget class="QLabel" name="label_3">
      <property name="layoutDirection">
       <enum>Qt::LeftToRight</enum>
      </property>
      <property name="text">
       <string>c</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
    </item>
    <item row="2" column="2" colspan="2">
     <widget class="QLineEdit" name="log_t"/>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>598</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionReset"/>
    <addaction name="actionSave"/>
    <addaction name="actionSave_as"/>
   </widget>
   <widget class="QMenu" name="menuImageProcesser">
    <property name="title">
     <string>ImageProcesser</string>
    </property>
    <addaction name="actionGray"/>
    <addaction name="actionReverse"/>
    <addaction name="actionContrast_stretch"/>
    <addaction name="actionHisteq"/>
    <addaction name="actionExact_Histeq"/>
    <addaction name="actionSpatial_Filter"/>
    <addaction name="actionMedian_Filter"/>
    <addaction name="actionTransform"/>
   </widget>
   <widget class="QMenu" name="menuFft2">
    <property name="title">
     <string>Fourier</string>
    </property>
    <addaction name="actionFft2"/>
    <addaction name="actionIfft2"/>
   </widget>
   <widget class="QMenu" name="menuMorphology">
    <property name="title">
     <string>Morphology</string>
    </property>
    <addaction name="actionErosion"/>
    <addaction name="actionDilation"/>
    <addaction name="actionOpening"/>
    <addaction name="actionClosing"/>
   </widget>
   <widget class="QMenu" name="menuEffects">
    <property name="title">
     <string>Effects</string>
    </property>
    <addaction name="actionGaussianBlur"/>
    <addaction name="actionEdgeExtraction"/>
    <addaction name="actionEmboss"/>
    <addaction name="actionOilPainting"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuImageProcesser"/>
   <addaction name="menuFft2"/>
   <addaction name="menuMorphology"/>
   <addaction name="menuEffects"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionLast"/>
   <addaction name="actionNext"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionOpen">
   <property name="text">
    <string>Open</string>
   </property>
  </action>
  <action name="actionReset">
   <property name="text">
    <string>Reset</string>
   </property>
  </action>
  <action name="actionLast">
   <property name="text">
    <string>上一张</string>
   </property>
   <property name="toolTip">
    <string>上一张</string>
   </property>
  </action>
  <action name="actionNext">
   <property name="text">
    <string>下一张</string>
   </property>
   <property name="toolTip">
    <string>下一张</string>
   </property>
  </action>
  <action name="actionGray">
   <property name="text">
    <string>Gray</string>
   </property>
  </action>
  <action name="actionReverse">
   <property name="text">
    <string>Reverse</string>
   </property>
  </action>
  <action name="actionContrast_stretch">
   <property name="text">
    <string>Contrast stretch</string>
   </property>
  </action>
  <action name="actionHisteq">
   <property name="text">
    <string>Histeq</string>
   </property>
  </action>
  <action name="actionExact_Histeq">
   <property name="text">
    <string>Exact_Histeq</string>
   </property>
  </action>
  <action name="actionSpatial_Filter">
   <property name="text">
    <string>Spatial_Filter</string>
   </property>
  </action>
  <action name="actionMedian_Filter">
   <property name="text">
    <string>Median_Filter</string>
   </property>
  </action>
  <action name="actionFft2">
   <property name="text">
    <string>fft2</string>
   </property>
  </action>
  <action name="actionIfft2">
   <property name="text">
    <string>ifft2</string>
   </property>
  </action>
  <action name="actionTransform">
   <property name="text">
    <string>Transform</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
  </action>
  <action name="actionErosion">
   <property name="text">
    <string>Erosion</string>
   </property>
  </action>
  <action name="actionDilation">
   <property name="text">
    <string>Dilation</string>
   </property>
  </action>
  <action name="actionOpening">
   <property name="text">
    <string>Opening</string>
   </property>
  </action>
  <action name="actionClosing">
   <property name="text">
    <string>Closing</string>
   </property>
  </action>
  <action name="actionGaussianBlur">
   <property name="text">
    <string>GaussianBlur</string>
   </property>
  </action>
  <action name="actionEdgeExtraction">
   <property name="text">
    <string>EdgeExtraction</string>
   </property>
  </action>
  <action name="actionEmboss">
   <property name="text">
    <string>Emboss</string>
   </property>
  </action>
  <action name="actionOilPainting">
   <property name="text">
    <string>OilPainting</string>
   </property>
  </action>
  <action name="actionSave_as">
   <property name="text">
    <string>Save_as</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
